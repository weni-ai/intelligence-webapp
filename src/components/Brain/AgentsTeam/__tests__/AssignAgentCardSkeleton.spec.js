import { mount } from '@vue/test-utils';

import AssignAgentSkeleton from '../AssignAgentCardSkeleton.vue';

describe('AgentCardSkeleton.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AssignAgentSkeleton);
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders skeleton loading components', () => {
    const skeletonLoadings = wrapper.findAll(
      '[data-testid="skeleton-loading"]',
    );
    expect(skeletonLoadings.length).toBe(3);
  });

  it('renders skills section', () => {
    const skillsSection = wrapper.find('[data-testid="skills-section"]');
    expect(skillsSection.exists()).toBe(true);
  });

  it('renders skills skeleton loadings', () => {
    const skillsSkeletonLoadings = wrapper.findAll(
      '[data-testid="skills-skeleton-loading"]',
    );
    expect(skillsSkeletonLoadings.length).toBe(2);
  });
});
