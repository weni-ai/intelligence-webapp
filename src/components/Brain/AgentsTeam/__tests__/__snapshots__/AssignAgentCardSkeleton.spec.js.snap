// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentCardSkeleton.vue > renders correctly 1`] = `
<section
  class="assign-agent-card-skeleton"
  data-testid="assign-agent-card-skeleton"
  data-v-e7b21008=""
>
  <div
    data-testid="skeleton-loading"
    data-v-e7b21008=""
  >
    
    <span
      class="unnnic-skeleton"
      style="background-color: rgb(232, 244, 244); background-image: linear-gradient( 90deg, #E8F4F4, rgba(255,255,255,0.375), #E8F4F4 ); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 30%; height: 22px;"
    >
       ‌ 
    </span>
    
  </div>
  <div
    data-testid="skeleton-loading"
    data-v-e7b21008=""
  >
    
    <span
      class="unnnic-skeleton"
      style="background-color: rgb(232, 244, 244); background-image: linear-gradient( 90deg, #E8F4F4, rgba(255,255,255,0.375), #E8F4F4 ); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 100%; height: 44px;"
    >
       ‌ 
    </span>
    
  </div>
  <section
    class="assign-agent-card-skeleton__skills"
    data-testid="skills-section"
    data-v-e7b21008=""
  >
    <div
      data-testid="skills-skeleton-loading"
      data-v-e7b21008=""
    >
      
      <span
        class="unnnic-skeleton"
        style="background-color: rgb(232, 244, 244); background-image: linear-gradient( 90deg, #E8F4F4, rgba(255,255,255,0.375), #E8F4F4 ); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 100px; height: 30px;"
      >
         ‌ 
      </span>
      
    </div>
    <div
      data-testid="skills-skeleton-loading"
      data-v-e7b21008=""
    >
      
      <span
        class="unnnic-skeleton"
        style="background-color: rgb(232, 244, 244); background-image: linear-gradient( 90deg, #E8F4F4, rgba(255,255,255,0.375), #E8F4F4 ); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 150px; height: 30px;"
      >
         ‌ 
      </span>
      
    </div>
  </section>
  <div
    data-testid="skeleton-loading"
    data-v-e7b21008=""
  >
    
    <span
      class="unnnic-skeleton"
      style="background-color: rgb(232, 244, 244); background-image: linear-gradient( 90deg, #E8F4F4, rgba(255,255,255,0.375), #E8F4F4 ); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 100%; height: 38px;"
    >
       ‌ 
    </span>
    
  </div>
</section>
`;
